### PR TITLE
Feat: Add SignUp page UI

### PR DESCRIPTION
**Pull request by nicktruong**
### Task
- [Authentication](https://trello.com/c/Phxj0MtZ)

### Describe
- Create SignUp page with 3 steps:
  - EmailForm: validate and collect user email
  - PasswordForm: validate and collect user password
  - UserInfoForm: validate and collect username, dob, and gender
- Validation is created with `joi` and `react-hook-form`

### Screenshots
- EmailForm
image

- PasswordForm
image

- UserInfoForm
image

### CODE REVIEW

File: src/common/utils/date-helpers.ts
Comment (carl-huynh-goldenowl) [inline src/common/utils/date-helpers.ts:1]: Should use camelCase for this file name

File: src/common/utils/date-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export function daysInMonth(month: number, year: number) {
```
Comment (carl-huynh-goldenowl) [inline src/common/utils/date-helpers.ts:1]: Should give another name to make the meaning clearer, start with verb

**Review by carl-huynh-goldenowl — APPROVED** [review]
Please update in next PR

**Comment by nicktruong** [comment]
All comments will be resolved as soon as possible

**Pull request merged**